language-guides/spring-data-neo4j/spring-data-neo4j.adoc

```diff
@@ -34,129 +35,194 @@ When developing with Neo4j, please use JDK 8 and your favorite IDE.
+   private List<Movie> movies = new ArrayList<>();
+}
+
 @NodeEntity
 public class Movie {
+   
+   @Id @GeneratedValue private Long id;
+   private String title;
+   private int released;
+   private String tagline;
+
+   @JsonIgnoreProperties("movie")
+   @Relationship(type = "ACTED_IN", direction = Relationship.INCOMING)
+   private List<Role> roles;
+}
 
-  @Id @GeneratedValue Long id;
+@RelationshipEntity(type = "ACTED_IN")
+public class Role {
 
-  String title;
+    @Id @GeneratedValue private Long id;
+	private List<String> roles = new ArrayList<>();
 
-  Person director;
+	@StartNode
+	private Person person;
 
-  @Relationship(type="ACTED_IN", direction = Relationship.INCOMING)
-  Set<Person> actors = new HashSet<>();
-}
+	@EndNode
+	private Movie movie;
 ----
 
+The repository interface allows the developer to create methods and queries needed for retrieving the data from the database.
+The interface extends the `Neo4jRepository`, which extends the Spring CRUDRepository for persisting and retrieving data.
+Some queries can be derived for us (like the `findByTitle` method), but others must be specifically written and annotated with `@Query`.
+
+There are also a lot of other methods available by default, such as `findOne`, `findAll`, etc.
+For more information, see the Spring Data Neo4j page and choose the latest https://projects.spring.io/spring-data-neo4j/[documentation link^] on the right side of the page.
+
 .Declare a repository interface
 [source,java]
 ----
-interface MovieRepository extends Neo4jRepository<Movie, Long> {
+public interface MovieRepository extends Neo4jRepository<Movie, Long> {
 
-  @Query("MATCH (m:Movie)<-[rating:RATED]-(user)
-          WHERE id(m) = {movieId} RETURN rating")
-  List<Rating> getRatings(@Param("movieID") Long movieId);
+    Movie findByTitle(@Param("title") String title);
 
-  List<Movie> findByTitle(String title);
+	Collection<Movie> findByTitleLike(@Param("title") String title);
+
+    @Query("MATCH (m:Movie)<-[r:ACTED_IN]-(a:Person) RETURN m,r,a LIMIT {limit}")
+	Collection<Movie> graph(@Param("limit") int limit);
 }
 ----
 
+The next section of code uses our repository finders in the `MovieService` class.
+The class starts by annotating it as a regular service with `@Service`. 
+Then, the `MovieService` class injects the `MovieRepository` using a constructor.
+
+Annotating these methods with `@Transactional` and read-only allows the smart client to efficiently route and execute the statements.
+In each method, the specified method is queried from the `MovieRepository` interface and results are returned to the service caller.
+
 .Use repository
 [source,java]
 ----
-@Autowired MovieRepository repo;
+@Service
+public class MovieService {
 
-List<Movie> movie = repo.findByTitle("The Matrix");
+    private final MovieRepository movieRepository;
+    public MovieService(MovieRepository movieRepository) {
+        this.movieRepository = movieRepository;
+    }
+
+    @Transactional(readOnly = true)
```

Comment:
same here, leave one of the `findByTitle*` variants off or find a better one (like the old rating one), otherwise, people will feel we waste their time ...